### PR TITLE
Moving ChannelFilterStore to noisepy-io

### PR DESCRIPTION
**Pull request by IshikaKhandelwal**
Added channel_filter_store.py and test_channel_filter_store.py to `noisepy-seis-io` and deleted from `noisepy-seis`

### CODE REVIEW

File: src/noisepy/seis/io/channelcatalog.py [hunk 1]
```diff
@@ -304,3 +304,11 @@ def cc_parameters(cc_para, coor, tcorr, ncorr, comp):
 
 
 # TODO: A channel catalog that uses the files in the SCEDC S3 bucket: s3://scedc-pds/FDSNstationXML/
+
```
Comment (carlosgjs) [inline src/noisepy/seis/io/channelcatalog.py:307]: It would be better not to have test code in the main package. My suggestion would be to leave this in `test_channelcatalog.py` and use it in all the `noisepy-seis-io` tests. For the `test_cross_correlation` in `noisepy-seis` you can create a mock, similar to how we are mocking the store [here](https://github.com/mdenolle/NoisePy/blob/136e8a6e7e24639ef8b68e8ea8277ee3af65314e/tests/test_cross_correlation.py#L47):

File: src/noisepy/seis/io/scedc_s3store.py [hunk 2]
```diff
@@ -175,3 +175,8 @@ def _parse_channel(filename: str) -> Channel:
             # lat/lon/elev will be populated later
             Station(network, station, location=location),
         )
+
+
+timespan1 = DateTimeRange(
```
Comment (carlosgjs) [inline src/noisepy/seis/io/scedc_s3store.py:180]: Similar to the above, let's not put test code in the package. I think the both `channelfilterstore.py` and `test_channelfilterstore.py` should move to `noisepy-seis-io` since they are both IO related.

**Review by carlosgjs — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -1,5 +1,10 @@
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
 [project]
-name = "noisepy-seis-io"
+name = "noisepy_seis_io"
```
[the remark this answers is not in the document]
Comment (IshikaKhandelwal) [reply pyproject.toml:6]: I am not sure why it appeared this way, because I have initialized the name as noisepy-seis-io only, will create a new PR with the changes mentioned and check again if it appears this way.

File: docs/conf.py [hunk 4]
```diff
@@ -17,10 +17,10 @@
 # -- Project information -----------------------------------------------------
 # https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
 
-project = "noisepy-seis-io"
+project = "noisepy_seis_io"
 copyright = "2023, Ishika Khandelwal"
 author = "Ishika Khandelwal"
-release = version("noisepy-seis-io")
+release = version("noisepy_seis_io")
```
Comment (carlosgjs) [inline docs/conf.py:23]: same question about the rename ...

File: .copier-answers.yml [hunk 5]
```diff
@@ -12,7 +12,7 @@ mypy_type_checking: none
 package_name: noisepy_seis_io
 preferred_linter: black
 project_license: MIT
-project_name: noisepy-seis-io
+project_name: noisepy_seis_io
```
Comment (carlosgjs) [inline .copier-answers.yml:15]: same question about the rename ...

File: src/noisepy/seis/io/utils.py [hunk 6]
```diff
@@ -67,6 +69,13 @@ def io_retry(func: Callable, *args, **kwargs) -> Any:
             time.sleep(FIND_RETRY_SLEEP * i)
 
 
+def date_range(month: int, start_day: int, end_day: int, start_hr: int = 0, end_hr: int = 0):
```
Comment (carlosgjs) [inline src/noisepy/seis/io/utils.py:72]: this is also test-only code, so it just be with the tests, not the production code. I would keep it in tests/utils.py. If you need it in `noisepy-seis` you could duplicate it there. Why duplication is generally not great in this case it's the least of two evils and this is a very simple piece of test only code.

File: tests/test_channelcatalog.py [hunk 7]
```diff
@@ -80,3 +72,11 @@ def test_XMLStationChannelCatalogCustomPath():
     yaq_inv = cat.get_inventory(DateTimeRange(), Station("CI", "YAQ"))
     assert len(yaq_inv) == 1
     assert len(yaq_inv.networks[0].stations) == 1
+
+
+class MockCatalog(ChannelCatalog):
```
Comment (carlosgjs) [inline tests/test_channelcatalog.py:77]: Is the code just moved? If so it's better to undo the change so you don't add to the file change history unnecessarily. And in general the fewer the changes the easier your PRs will be to review.

**Review by carlosgjs — CHANGES_REQUESTED** [review]

File: tests/test_channelcatalog.py (hunk 7 above)
Comment (carlosgjs) [reply tests/test_channelcatalog.py:77]: @IshikaKhandelwal can you revert this change? It's not needed, right?

**Review by carlosgjs — CHANGES_REQUESTED** [review]

File: tests/test_channelcatalog.py (hunk 7 above)
Comment (IshikaKhandelwal) [reply tests/test_channelcatalog.py:77]: It's needed. `test_channel_filter_store.py` and `test_scedc_s3store.py` in `noisepy-seis-io` imports MockCatalog.

**Review by carlosgjs — APPROVED** [review]
Great, now the PR looks very clean and simple. In fact, I suggest renaming the PR to: "Moving LocationChannelFilterStore to noisepy-io" since that's effectively what it boiled down to.
